Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/LoginFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.cmput3owo1.moodlet.fragments;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.cmput3owo1.moodlet.R;
+import com.cmput3owo1.moodlet.activities.MainActivity;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+public class LoginFragment extends Fragment {
+
+    EditText email, password;
+    TextView signupText;
+    Button loginButton;
+
+    FirebaseAuth auth;
+
+    public LoginFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        auth = FirebaseAuth.getInstance();
+
+        if(auth.getCurrentUser() != null){
+            Intent intent = new Intent(getActivity(), MainActivity.class);
+            startActivity(intent);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        // Inflate the layout for this fragment
+        View loginFragmentView = inflater.inflate(R.layout.fragment_login, container, false);
+
+        signupText = loginFragmentView.findViewById(R.id.sign_up_text);
+        email = loginFragmentView.findViewById(R.id.edit_text_email);
+        password = loginFragmentView.findViewById(R.id.edit_text_password);
+        loginButton = loginFragmentView.findViewById(R.id.btn_login);
+
+        signupText.setOnClickListener(new View.OnClickListener() {
```

Comment:
I think it's cleaner to add an "onSignupTestClick" function to the fragment and then in the XML, set the onClick property to the function

Reply:
I'll do that in a separate PR